core/validator.go

```diff
@@ -87,12 +88,20 @@ func (v *shardValidator) UpdateOperatorID(operatorID OperatorID) {
 }
 
 func (v *shardValidator) ValidateBatch(batchHeader *BatchHeader, blobs []*BlobMessage, operatorState *OperatorState, pool common.WorkerPool) error {
-
-	err := validateBatchHeaderRoot(batchHeader, blobs)
+	headers := make([]*BlobHeader, len(blobs))
+	for i, blob := range blobs {
+		headers[i] = blob.BlobHeader
+	}
+	err := ValidateBatchHeaderRoot(batchHeader, headers)
 	if err != nil {
 		return err
 	}
 
+	return v.ValidateBlobs(blobs, operatorState, pool)
```

Comment:
This is a all or none check ? I.e either all blobs are valid or error is returned ?

Reply:
Yes, it's all or none check for now. 
Down the road we may filter out invalid blobs from the set and only sign on valid blobs